pkg/shared/constants/compare_contracts_test.go

```diff
@@ -0,0 +1,55 @@
+package constants_test
+
+import (
+	"github.com/makerdao/vdb-transformer-utilities/pkg/fakes"
+	"github.com/makerdao/vdb-transformer-utilities/pkg/shared/constants"
+	"github.com/makerdao/vulcanizedb/pkg/eth"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Compare ABIs", func() {
+	BeforeEach(func() {
+		fakes.SetFakeConfig()
+	})
+
+	It("has no error if the ABIs are equal", func() {
+		flipEthABI, parseErr := eth.ParseAbi(constants.GetContractABI("MCD_FLIP_ETH_A_1.0.0"))
+		Expect(parseErr).NotTo(HaveOccurred())
+		flipBatABI, parseErr := eth.ParseAbi(constants.GetContractABI("MCD_FLIP_BAT_A_1.0.0"))
+		Expect(parseErr).NotTo(HaveOccurred())
+
+		Expect(constants.CompareContractABI(flipEthABI, flipBatABI)).To(Succeed())
+	})
+
+	It("has a constructor error if ABIs differ on the constructor", func() {
+		medianBalABI, parseErr := eth.ParseAbi(constants.GetContractABI("MEDIAN_BAL"))
+		Expect(parseErr).NotTo(HaveOccurred())
+		medianBatABI, parseErr := eth.ParseAbi(constants.GetContractABI("MEDIAN_BAT"))
+		Expect(parseErr).NotTo(HaveOccurred())
+
+		expectedError := constants.NewMismatchedConstructorsError(medianBalABI, medianBatABI)
+		Expect(constants.CompareContractABI(medianBalABI, medianBatABI)).To(MatchError(expectedError))
+	})
+
+	It("has a method error if the ABIs have the same constructor but differing methods", func() {
+		medianBalABI, parseErr := eth.ParseAbi(constants.GetContractABI("MEDIAN_BAL"))
+		Expect(parseErr).NotTo(HaveOccurred())
+		medianBalWithoutDenyABI, parseErr := eth.ParseAbi(constants.GetContractABI("MEDIAN_BAL_WITHOUT_DENY"))
+		Expect(parseErr).NotTo(HaveOccurred())
+
+		expectedError := constants.NewMismatchedMethodsError(medianBalABI, medianBalWithoutDenyABI, "deny")
+		Expect(constants.CompareContractABI(medianBalABI, medianBalWithoutDenyABI)).To(MatchError(expectedError))
+	})
+
+	It("has an event error if the ABIs have the same constructor and methods but differing events", func() {
+		medianBalABI, parseErr := eth.ParseAbi(constants.GetContractABI("MEDIAN_BAL"))
+		Expect(parseErr).NotTo(HaveOccurred())
+		medianBalWithoutLogMedianPriceABI, parseErr := eth.ParseAbi(constants.GetContractABI("MEDIAN_BAL_WITHOUT_LOG_MEDIAN_PRICE"))
+		Expect(parseErr).NotTo(HaveOccurred())
+
+		expectedError := constants.NewMismatchedEventsError(medianBalABI, medianBalWithoutLogMedianPriceABI, "LogMedianPrice")
+		Expect(constants.CompareContractABI(medianBalABI, medianBalWithoutLogMedianPriceABI)).To(MatchError(expectedError))
```

Comment:
Not sure it's necessary, but do you think it would be helpful to add test cases for if the methods/events exist in both ABIs, but are different? I think these last two tests are only checking the case where the method/event doesn't exist in one of the ABIs.

Reply:
Probably not a bad idea but I don't think this commit is supposed to be in this PR. I think it's a holdover from a previous one that will go away on rebase.